### PR TITLE
ipe: update to version 7.2.29

### PR DESCRIPTION
**Pull request by jjstickel**
* update ipe to version 7.2.29
* update ipe-tools to 7.2.24.1
* remove qt4 variant
* add ipetoipe to $prefix/bin
* add jjstickel to maintainer

Closes: https://trac.macports.org/ticket/64422
Closes: https://trac.macports.org/ticket/60911
Closes: https://trac.macports.org/ticket/68666

#### Description

Ipe has been neglected in Macports. This is my attempt to get Ipe and Ipe-tools working via macports install. I used the Portfile from https://trac.macports.org/ticket/64422 as a starting point (thanks to jxy), but I left out Ipe-presenter. I feel like Ipe-presenter should be it's own port or a subport. Further contributions are welcome.

Note that the `qt5` does not build. I recommend removing the variant completely, as it looks like it really isn't expected by the upstream developer to be used on Mac, and the bundle works well. If others agree, I can remove the variant and commit that change as well.

Note further that `pdftoipe` failed to compile, and so it is currently commented-out in the `ipe-tools` Portfile. Again, help from others is welcome.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.6.1 23G93 arm64
Xcode 15.4 15F31d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by maehne — COMMENTED** [review]
I unfortunately don't have the bandwidth anymore to maintain this and other ports, as macOS is not anymore my primary work platform. Nevertheless, your PR looks mostly good to me. You may consider to update it to Qt6 though. Also, the compile error for `pdftoipe` should be relatively easy to fix. My guess is that it is caused by `poppler`, which requires C++'20. It may be worthwhile to study the `PKGBUILD` and associated patches of [Ipe](https://aur.archlinux.org/packages/ipe) and [ipe-tools](https://aur.archlinux.org/packages/ipe-tools-git) in the AUR to improve further these ports.

**Comment by jjstickel** [comment]
> I unfortunately don't have the bandwidth anymore to maintain this and other ports, as macOS is not anymore my primary work platform. Nevertheless, your PR looks mostly good to me. You may consider to update it to Qt6 though. Also, the compile error for `pdftoipe` should be relatively easy to fix. My guess is that it is caused by `poppler`, which requires C++'20. It may be worthwhile to study the `PKGBUILD` and associated patches of [Ipe](https://aur.archlinux.org/packages/ipe) and [ipe-tools](https://aur.archlinux.org/packages/ipe-tools-git) in the AUR to improve further these ports.

Thanks for the input. I really can't spend much more time on these, certainly not what it would take to add a `qt6` variant. I'm hoping to see some input from active developers.

**Comment by herbygillot** [comment]
Thanks for this submission. Some comments follow:

File: graphics/ipe-tools/Portfile [hunk 1]
```diff
@@ -20,18 +19,20 @@ long_description        The following tools for the Ipe extensible drawing \
 
 homepage                http://ipe.otfried.org/
```
Comment (herbygillot) [inline graphics/ipe-tools/Portfile:20]: Can we make this URL HTTPS? Ideally all URLs should be.

File: graphics/ipe-tools/Portfile [hunk 2]
```diff
@@ -20,18 +19,20 @@ long_description        The following tools for the Ipe extensible drawing \
...
 
-checksums               sha1    a650856f7123fe68499f20bbe66695cb040306f7 \
-                        rmd160  9ee6d4eb84455d55a36d24e2233de9f6a25a21f7
+checksums           sha1    467cf36177a2fc6b6daf3410f43733bdff071c58 \
```
Comment (herbygillot) [inline graphics/ipe-tools/Portfile:22]: The `sha1` checksum is no longer part of the recommended set - we can delete this line.

File: graphics/ipe-tools/Portfile [hunk 3]
```diff
@@ -20,18 +19,20 @@ long_description        The following tools for the Ipe extensible drawing \
...
+                    size    239310
 
 patchfiles              patch-svgtoipe-svgtoipe.py.diff
 post-patch {
-    reinplace "s|__MP_PYTHON_BIN__|${prefix}/bin/python2.7|" ${worksrcpath}/svgtoipe/svgtoipe.py
+    reinplace "s|__MP_PYTHON_BIN__|${prefix}/bin/python3|" ${worksrcpath}/svgtoipe/svgtoipe.py
 }
 
-set python.pkgd ${frameworks_dir}/Python.framework/Versions/2.7/lib/python2.7/site-packages
+set python.pkgd ${frameworks_dir}/Python.framework/Versions/3.11/lib/python3.11/site-packages
```
Comment (herbygillot) [inline graphics/ipe-tools/Portfile:32]: Is there a reason why we prefer Python 3.11 over 3.12?

File: graphics/ipe/Portfile [hunk 4]
```diff
@@ -18,40 +19,27 @@ long_description        \
...
 platforms               macosx
-master_sites            https://dl.bintray.com/otfried/generic/ipe/${branch}
-distname                ${name}-${version}-src
+
 worksrcdir              ${name}-${version}/src
 
-checksums               rmd160  eaa2b9fc1a923a78b14e1c1c27f8ffc139febe0e \
-                        sha256  a22dcae9cb660f466678a6b568e9fed1b12a3830e49465594d65fc789b0ba725 \
-                        size    1950071
+checksums           rmd160  3aab1c281c938f10c80d29d23c46df963c645b8c \
```
Comment (herbygillot) [inline graphics/ipe/Portfile:25]: Can we match the alignment that was there previously?

**Comment by jonCAstickel** [comment]
> Thanks for this submission. Some comments follow:

Thanks for the comments. I'll update the PR when I have a few minutes next week. Any input on removing the QT5 variant?

**Comment by jjstickel** [comment]
> Thanks for this submission. Some comments follow:

Thanks for the comments. I'll update the PR when I have a few minutes next week. Any input on removing the QT5 variant?

File: graphics/ipe-tools/Portfile (hunk 1 above)
Comment (jjstickel) [reply graphics/ipe-tools/Portfile]: done

File: graphics/ipe/Portfile (hunk 4 above)
Comment (jjstickel) [reply graphics/ipe/Portfile]: done

**Comment by herbygillot** [comment]
Looks good, thanks for your work on this, @jjstickel.

**Pull request merged**